src/superface-test.utils.ts

```diff
@@ -0,0 +1,369 @@
+  payload: {
+    definitions?: RecordingDefinition[];
+    scopes?: RecordingScope[];
+  };
+}): void {
+  if (payload.definitions) {
+    for (const definition of payload.definitions) {
+      for (const scheme of securitySchemes) {
+        const securityValue = securityValues.find(val => val.id === scheme.id);
+        removeCredentials({ definition, scheme, securityValue, baseUrl });
+      }
+    }
+  }
+
+  if (payload.scopes) {
+    for (const scope of payload.scopes) {
+      for (const scheme of securitySchemes) {
+        const securityValue = securityValues.find(val => val.id === scheme.id);
+        loadCredentials({ scope, scheme, securityValue });
+      }
+    }
+  }
+}
+
+export function loadCredentials({
+  scope,
+  scheme,
+  securityValue,
+}: {
+  scope: RecordingScope;
+  scheme: SecurityScheme;
+  securityValue?: SecurityValues;
+}): void {
+  if (isApiKeySecurityScheme(scheme)) {
+    const schemeName = scheme.name ?? AUTH_HEADER_NAME;
+    let loadedCredential: string | undefined;
+
+    if (isApiKeySecurityValues(securityValue)) {
+      if (securityValue.apikey.startsWith('$')) {
+        loadedCredential = process.env[securityValue.apikey.substr(1)];
+      } else {
+        loadedCredential = securityValue.apikey;
+      }
+    }
+
+    if (scheme.in === ApiKeyPlacement.BODY) {
+      if (loadedCredential !== undefined) {
+        scope.filteringRequestBody(
+          new RegExp(loadedCredential, 'g'),
+          HIDDEN_CREDENTIALS_PLACEHOLDER
+        );
+      }
+    } else if (scheme.in === ApiKeyPlacement.QUERY) {
+      scope.filteringPath(
+        new RegExp(schemeName + '([^&#]+)', 'g'),
+        `${schemeName}=${HIDDEN_CREDENTIALS_PLACEHOLDER}`
+      );
+    } else if (scheme.in === ApiKeyPlacement.PATH) {
+      if (loadedCredential !== undefined) {
+        scope.filteringPath(
+          new RegExp(loadedCredential, 'g'),
+          HIDDEN_CREDENTIALS_PLACEHOLDER
+        );
+      }
+    }
+  } else if (isDigestSecurityScheme(scheme)) {
+    throw new UnexpectedError('not implemented');
+  }
+}
+
+export function removeCredentials({
+  definition,
+  scheme,
+  securityValue,
+  baseUrl,
+}: {
+  definition: RecordingDefinition;
+  scheme: SecurityScheme;
+  securityValue?: SecurityValues;
+  baseUrl: string;
+}): void {
+  if (isApiKeySecurityScheme(scheme)) {
+    let loadedCredential: string | undefined;
+
+    if (isApiKeySecurityValues(securityValue)) {
+      if (securityValue.apikey.startsWith('$')) {
+        loadedCredential = process.env[securityValue.apikey.substr(1)];
+      } else {
+        loadedCredential = securityValue.apikey;
+      }
+    }
+
+    if (scheme.in === ApiKeyPlacement.HEADER) {
+      if (scheme.name !== undefined) {
+        if (definition.reqheaders?.[scheme.name] !== undefined) {
+          definition.reqheaders[scheme.name] = HIDDEN_CREDENTIALS_PLACEHOLDER;
+        }
+      } else {
+        if (definition.reqheaders?.['X-API-KEY'] !== undefined) {
```

Comment:
why have this particular header here?

Reply:
I refactored it, [now it looks](https://github.com/superfaceai/testing-lib/blob/fedf655d745d52e6d26dad5cc125573ad10e0ad8/src/nock.utils.ts#L38-L49) for loaded environment variable in header values. I also created mutation functions for each case and moved it to `nock.utils.ts`.